typing/printtyp.ml

```diff
@@ -2176,15 +2176,14 @@ let prepare_trace f tr =
 (** Keep elements that are not [Diff _ ] and take the decision
     for the last element, require a prepared trace *)
 let rec filter_trace
-          (trace_format : 'variety trace_format)
           keep_last
   : ('a, 'variety) Errortrace.t -> _ = function
```

Comment:
I would propose to also remove the type annotation above: the original motivation for the `trace_format` parameter was to ensure that the `filter_trace` function was called on the expected variant of error traces.